File: src/app/_modules/service.ts
import { Contract } from "./contract";
import { User } from "./user";

export class Service {
    constructor(
        public userId : number,
        public title : string,
        public description : string,
        public price : number,
        public category : string,
        public status : string,
        public createdDate : Date,
        public serviceId? : number,
        public user? : User,
        public contract? :Contract
    // "serviceId": 0,
    // "userId": 0,
    // "title": "string",
    // "description": "string",
    // "price": 0,
    // "category": "string",
    // "status": "string",
    // "createdDate": "2024-10-15T15:44:56.219Z",
    ){}
}

File: Backend/DataAccess/Models/Service.cs
using System.ComponentModel.DataAnnotations;
using System.Collections.Generic;
using Backend.BusinessLogic;

namespace Backend.Models
{
    public class Service
    {
        [Required]
        public int ServiceId { get; set; }

        public int UserId { get; set; }

        public string? Title { get; set; }

        public string? Description { get; set; }

        public double Price { get; set; }

        public string? Category { get; set; }

        public ServiceStatus Status { get; set; } = ServiceStatus.Pending;


        public DateTime CreatedDate { get; set; }

        public virtual Contract? Contract { get; set; }
        public virtual User? Freelancer { get; set; }

        
        public virtual ICollection<Proposal> Proposals { get; set; } = new List<Proposal>();
    }
}
